### ISSUE TITLE
ColorSpace must be TYPE_RGB

### ISSUE THREAD

**Issue by talhahasanzia** [android/size-analyzer#9]
**Describe the bug**
I am trying to run the analyzer from command line using `size-analyzer check-project` but it is crashing with an exception of ColorSpace.

**Size-Analyzer version(s) affected**
Version: 0.3.1

**Stacktrace**
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/D:/projects/analyzer/analyzer/bin/analyzer.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
3
5
AndroidPluginVersion{majorVersion=3, minorVersion=5}
Exception in thread "main" picocli.CommandLine$ExecutionException: Error while calling command (com.android.tools.sizereduction.analyzer.cli.CheckProject@345e5a17): java.lang.IllegalArgu
mentException: ColorSpace must be TYPE_RGB.
        at picocli.CommandLine.execute(CommandLine.java:1068)
        at picocli.CommandLine.access$900(CommandLine.java:142)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:1255)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:1223)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1131)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:1414)
        at picocli.CommandLine.parseWithHandler(CommandLine.java:1353)
        at com.android.tools.sizereduction.analyzer.AnalyzerMain.handleCommand(AnalyzerMain.java:66)
        at com.android.tools.sizereduction.analyzer.AnalyzerMain.main(AnalyzerMain.java:62)
Caused by: java.lang.IllegalArgumentException: ColorSpace must be TYPE_RGB.
        at java.desktop/java.awt.image.PackedColorModel.<init>(Unknown Source)
        at java.desktop/java.awt.image.DirectColorModel.<init>(Unknown Source)
        at org.apache.commons.imaging.ColorTools.deriveColorModel(ColorTools.java:122)
        at org.apache.commons.imaging.ColorTools.deriveColorModel(ColorTools.java:96)
        at org.apache.commons.imaging.ColorTools.deriveColorModel(ColorTools.java:91)
        at org.apache.commons.imaging.ColorTools.relabelColorSpace(ColorTools.java:71)
        at org.apache.commons.imaging.ColorTools.convertBetweenColorSpaces(ColorTools.java:239)
        at org.apache.commons.imaging.formats.png.PngImageParser.getBufferedImage(PngImageParser.java:650)
        at org.apache.commons.imaging.Imaging.getBufferedImage(Imaging.java:1375)
        at org.apache.commons.imaging.Imaging.getBufferedImage(Imaging.java:1269)
        at org.apache.commons.imaging.Imaging.getBufferedImage(Imaging.java:1238)
        at com.android.tools.sizereduction.analyzer.suggesters.binaryfiles.WebpSuggester.safelyParseImage(WebpSuggester.java:133)
        at com.android.tools.sizereduction.analyzer.suggesters.binaryfiles.WebpSuggester.processFileEntry(WebpSuggester.java:86)
        at com.android.tools.sizereduction.analyzer.suggesters.binaryfiles.WebpSuggester.processProjectEntry(WebpSuggester.java:72)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:112)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:97)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:97)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:97)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:97)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeProject(ProjectAnalyzer.java:71)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeDirectory(ProjectAnalyzer.java:93)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyzeProject(ProjectAnalyzer.java:71)
        at com.android.tools.sizereduction.analyzer.analyzers.ProjectAnalyzer.analyze(ProjectAnalyzer.java:53)
        at com.android.tools.sizereduction.analyzer.cli.CheckProject.call(CheckProject.java:96)
        at com.android.tools.sizereduction.analyzer.cli.CheckProject.call(CheckProject.java:37)
        at picocli.CommandLine.execute(CommandLine.java:1061)
        ... 8 more
```

**To Reproduce**
> size-analyzer check-project D:\...\project\

**Expected behavior**
It should analyze size of the project

**Known workaround**
None. 

**Environment:**
OS: Windows 10, AS 3.5.

**Additional context**
We have a separate dependencies file written in gradle. This file is included in main app (gradle).

**Comment by nissaFaheem** [comment]
Did you find any solution for it?

**Comment by svarnik** [comment]
This happens for me too.

**Comment by tdeck** [comment]
It looks like there's a problem with AWT not supporting various color spaces when parsing a PNG. I've tried to reproduce this with some of [these images](http://www.schaik.com/pngsuite/pngsuite_bas_png.html) and some of [these images](https://github.com/codelogic/wide-gamut-tests) but I haven't gotten it to throw an exception. Does anyone have a test image that they can share for a repro?

**Comment by tdeck** [comment]
This has been fixed in release 0.4.0. Although I wasn't able to repro it, we should be correctly swallowing these errors and passing over images that don't load.

I'm sorry it's taken so long to address this issue. The Size Analyzer was something of an experimental project, and we haven't had a team dedicated to continuing its development. Since I'm leaving Google in a couple of days, I thought I'd try to tie up these loose ends. I've also updated the README to reflect the status of the project.
